.github/actions/setup-k3d/action.yaml

```
@@ -72,30 +72,52 @@ runs:
       shell: bash
       run: |
         k3d version
-    - name: Setup k3d options
+    - name: Create k8s cluster
       shell: bash
-      id: k3d_options
+      id: start_k3d
       run: |
-        if [ "${AGENTS}" != 0 ]; then
-          OPTIONS="${OPTIONS} --agents ${AGENTS}"
+        if [ -z "${K3D_CLUSTER_NAME}" ]; then
+          K3D_CLUSTER_NAME="vald-cluster"
+        fi
+        if [ -z "${K3D_NODES}" ]; then
+          K3D_NODES=5
         fi
-        if [ "${INGRESS_PORT}" != 0 ]; then
-          OPTIONS="${OPTIONS} -p ${INGRESS_PORT}:80@loadbalancer"
+        if [ -z "${K3D_INGRESS_PORT}" ]; then
+          K3D_INGRESS_PORT=8081
```

```suggestion
        if [ -n "${K3D_INGRESS_PORT}" ]; then
          K3D_OPTIONS="${K3D_OPTIONS} -p ${K3D_INGRESS_PORT}:80@loadbalancer"
```

Since the default value of  ingress_port is 0, it also may be set to an empty character.